AWSCognitoAuth/Internal/UICKeyChainStore/AWSCognitoAuthUICKeyChainStore.m

```diff
@@ -933,6 +933,26 @@ + (NSArray *)prettify:(CFTypeRef)itemClass items:(NSArray *)items
 
 #pragma mark -
 
+- (void)migrateToCurrentAccessibility {
+    NSArray *items = [self allItems];
+    for (NSDictionary *item in items) {
+        CFComparisonResult result = CFStringCompare((CFStringRef)item[@"accessibility"],
+                                                    [self accessibilityObject], 0);
+        if (result == 0) {
+            continue;
+        }
+        NSString *key = item[@"key"];
+        NSObject *value = item[@"value"];
+        if ([value isKindOfClass:[NSString class]]) {
+            [self setString: (NSString *)value forKey:key];
```

Comment:
Do `setString` / `setData` write to Keychain **and** back to `allItems`?

Reply:
`allItems` is a getter that do `SecItemCopyMatching` to get all the items in the keychain under the current `service` string. So if you call `allItems` again you will see the new items inside. 

Here is a log I printed with key and accessibility:

Logs before `AccessibleAfterFirstUnlockThisDeviceOnly` was added:

```
User Pool

Optional(us-xxx.asf.device.id) Optional(ck)
Optional(us-xxx.asf.device.id) Optional(ck)
Optional(xx.currentUser) Optional(ck)
Optional(xx-x.idToken) Optional(ck)
Optional(xxx.accessToken) Optional(ck)
Optional(xxx.refreshToken) Optional(ck)
Optional(xxx.tokenExpiration) Optional(ck)
Optional(xxx.device.id) Optional(ck)
Optional(xx.device.secret) Optional(ck)
Optional(xx.device.group) Optional(ck)

AWSMC

Optional(configurationKey) Optional(ck)
Optional(loginsMap) Optional(ck)
Optional(federationProvider) Optional(ck)

Identity Pool

Optional(identityId) Optional(ck)
Optional(accessKey) Optional(ck)
Optional(secretKey) Optional(ck)
Optional(sessionKey) Optional(ck)
Optional(expiration) Optional(ck)
```

Logs after this PR was added: 

```
User Pool

Optional(us-xxx.asf.device.id) Optional(cku)
Optional(xx.currentUser) Optional(cku)
Optional(xxx.idToken) Optional(cku)
Optional(xxx.accessToken) Optional(cku)
Optional(xxx.refreshToken) Optional(cku)
Optional(xxx.tokenExpiration) Optional(cku)
Optional(xxx.device.id) Optional(cku)
Optional(xxx.device.secret) Optional(cku)
Optional(xxx.device.group) Optional(cku)

AWSMC

Optional(configurationKey) Optional(cku)
Optional(loginsMap) Optional(cku)
Optional(federationProvider) Optional(cku)

Identity Pool

Optional(identityId) Optional(cku)
Optional(accessKey) Optional(cku)
Optional(secretKey) Optional(cku)
Optional(sessionKey) Optional(cku)
Optional(expiration) Optional(cku)
```

Reply:
Perfect, thanks.